ctapipe/io/eventsource.py

```diff
@@ -3,11 +3,43 @@
 """
 from abc import abstractmethod
 from os.path import exists
-from traitlets import Unicode, Int, Set
-from ctapipe.core import Component
+from traitlets import Unicode, Int, Set, TraitError
+from ctapipe.core import Component, non_abstract_children
 from ctapipe.core import Provenance
+from traitlets.config.loader import LazyConfigValue
 
-__all__ = ['EventSource', ]
+__all__ = ['EventSource', 'event_source']
+
+
+def event_source(input_url, *args, **kwargs):
```

Comment:
I'm not sure what is gained by allowing positional arguments from event_source and from_config to be passed to EventSource. Perhaps only allow named arguments (kwargs) for both methods?